go-controller/pkg/ovn/logical_switch_cache.go

```diff
@@ -0,0 +1,168 @@
+package ovn
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	ipam "k8s.io/kubernetes/pkg/registry/core/service/ipallocator"
+
+	"reflect"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	"k8s.io/klog"
+)
+
+// logicalSwitchInfo contains information corresponding to the node. It holds the
+// subnet allocations (v4 and v6) as well as the IPAM allocator instances for each
+// subnet managed for this node
+// logicalSwitchInfo is used to save switch specific information such as subnets,
+// allocated IPs per switch etc.
+type logicalSwitchInfo struct {
+	hostSubnets []*net.IPNet
+	ipams       []ipam.Interface
+}
+
+type logicalSwitchCache struct {
+	cache map[string]logicalSwitchInfo
```

Comment:
@gvbalaji I would say leave as-is, because this is already a logicalSwitchCache object so there's not much point in being redundant? Especially if things outside this object are not supposed to touch internal members.

Reply:
ok. That's fine. I was only asking because in methods where structure pointer is also named cache,  references to cache.cache was not easy on the eyes :) But I am fine with your assessment.